pydra/engine/core.py

```diff
@@ -608,3 +617,14 @@ def is_task(obj):
 
 def is_workflow(obj):
     return isinstance(obj, Workflow)
+
+
+def is_runnable(graph, obj):
+    """Check if a task within a graph is runnable"""
+    if (not is_task(obj)) or not hasattr(graph, "predecessors"):
```

Comment:
I don't understand this if statement, and none of the tests are coming to this part. 

If you're trying to check that the obj is a `Workflow`, `not is_task(obj)` won't do the job

Reply:
`obj` can be a `Task` or `Workflow`

Reply:
so when do you expect to execute the statement in the `if` part?

Reply:
this can be removed, was essentially a catch for incorrect usage

Reply:
if this is the motivation, we should raise an exception, returning `False` is misleading